canonical-web-and-design/maas-ui
# feat(ui): build Navigation component. Update MAAS logo

## Done

- Create a reusable Vanilla Navigation component.
- Update the logo for MAAS.
- Upgrade Vanilla and react-components.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check that the main nav looks good at mobile, middle and large sizes (the mobile menu should be able to be opened and closed, the hardware menu appears at middle size).
- Check that you can navigate between pages.
- Go to the controller details (angular) and make sure the header works as it does from the React client.
- Check that you log out from controller details as well as from a React page.

## Fixes

Fixes: #3818.
Fixes: canonical-web-and-design/app-tribe#833.
Fixes: #3454.

## Screenshots

<img width="1309" alt="Screen Shot 2022-04-21 at 2 37 30 pm" src="https://user-images.githubusercontent.com/361637/164392418-7dd48678-87d7-4554-8145-2bc7aac5a9eb.png">
<img width="427" alt="Screen Shot 2022-04-21 at 2 37 53 pm" src="https://user-images.githubusercontent.com/361637/164392440-0138586a-fb52-4040-99ca-bd3d02002438.png">
<img width="1007" alt="Screen Shot 2022-04-21 at 2 37 39 pm" src="https://user-images.githubusercontent.com/361637/164392433-b63d3db8-1dbe-4183-9bf5-9c3d59ca5593.png">
